From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sun, 29 May 2016 11:37:08 +0200
Subject: ar71xx: Send power to USB port on WNR2200

This patch fixes ticket #15267 by enabling power on the
WNR2200's USB port. At present, the USB port on the WNR2200
is non-functional due to it not receiving power.

This patch defines an additional GPIO pin, but none of the
current GPIO definitions have been modified.

Signed-off-by: Riley Baird <BM-2cVqnDuYbAU5do2DfJTrN7ZbAJ246S4XiX@bitmessage.ch>

Backport of r47236

diff --git a/target/linux/ar71xx/files/arch/mips/ath79/mach-wnr2200.c b/target/linux/ar71xx/files/arch/mips/ath79/mach-wnr2200.c
index 0087cac..5d23f21 100644
--- a/target/linux/ar71xx/files/arch/mips/ath79/mach-wnr2200.c
+++ b/target/linux/ar71xx/files/arch/mips/ath79/mach-wnr2200.c
@@ -36,7 +36,7 @@
 #define WNR2200_GPIO_LED_LAN4_GREEN	16
 #define WNR2200_GPIO_LED_PWR_AMBER	21
 #define WNR2200_GPIO_LED_PWR_GREEN	22
-
+#define WNR2200_GPIO_USB_5V		4
 #define WNR2200_GPIO_USB_POWER		24
 
 #define WNR2200_KEYS_POLL_INTERVAL	20 /* msecs */
@@ -128,9 +128,9 @@ static void __init wnr2200_setup(void)
 					wnr2200_leds_gpio);
 
 	/* enable power for the USB port */
-	gpio_request_one(WNR2200_GPIO_USB_POWER,
-			GPIOF_OUT_INIT_HIGH | GPIOF_EXPORT_DIR_FIXED,
-			"USB power");
+	ap9x_pci_setup_wmac_gpio(0,
+		BIT(WNR2200_GPIO_USB_5V),
+		BIT(WNR2200_GPIO_USB_5V));
 
 	ath79_register_usb();
 }
